rasa/engine/recipes/default_recipe.py

```diff
@@ -816,3 +836,254 @@ def _add_end_to_end_features_for_inference(
+        else:
+            all_keys = rasa.shared.constants.CONFIG_KEYS
+
+        return {k for k in all_keys if k not in config.keys()}
+
+    @staticmethod
+    def complete_config(
+        config: Dict[Text, Any], keys_to_configure: Set[Text]
+    ) -> Dict[Text, Any]:
+        """Complete a config by adding automatic configuration for the specified keys.
+
+        Args:
+            config: The provided configuration.
+            keys_to_configure: Keys to be configured automatically (e.g. `policies`).
+
+        Returns:
+            The resulting configuration including both the provided and
+            the automatically configured keys.
+        """
+        import pkg_resources
+
+        if keys_to_configure:
+            logger.debug(
+                f"The provided configuration does not contain the key(s) "
+                f"{transform_collection_to_sentence(keys_to_configure)}. "  # noqa: E501, W505
+                f"Values will be provided from the default configuration."
+            )
+
+        filename = "config_files/default_config.yml"
+
+        default_config_file = pkg_resources.resource_filename(__name__, filename)
+        default_config = rasa.shared.utils.io.read_config_file(default_config_file)
+
+        config = copy.deepcopy(config)
+        for key in keys_to_configure:
+            config[key] = default_config[key]
+
+        return config
+
+    @staticmethod
+    def _dump_config(
+        config: Dict[Text, Any],
+        config_file_path: Text,
+        missing_keys: Set[Text],
+        auto_configured_keys: Set[Text],
+        training_type: Optional[TrainingType] = TrainingType.BOTH,
+    ) -> None:
+        """Dump the automatically configured keys into the config file.
+
+        The configuration provided in the file is kept as it is (preserving the order of
+        keys and comments).
+        For keys that were automatically configured, an explanatory
+        comment is added and the automatically chosen configuration is
+        added commented-out.
+        If there are already blocks with comments from a previous auto
+        configuration run, they are replaced with the new auto
+        configuration.
+
+        Args:
+            config: The configuration including the automatically configured keys.
+            config_file_path: The file into which the configuration should be dumped.
+            missing_keys: Keys that need to be added to the config file.
+            auto_configured_keys: Keys for which a commented out auto
+            configuration section needs to be added to the config file.
+            training_type: NLU, CORE or BOTH depending on which is trained.
+        """
+        config_as_expected = DefaultV1Recipe._is_config_file_as_expected(
+            config_file_path, missing_keys, auto_configured_keys, training_type
+        )
+        if not config_as_expected:
+            rasa.shared.utils.cli.print_error(
```

Comment:
Uh, this looks weird. Are we sure we don't want to raise here? Like if I for whatever reason managed to move my config back to it belongs (without noticing because I copy stuff back and forth and retrain and then get a different result 🤔

Reply:
I think this is a very edge case, I won't imagine anyone changing config file while config is being read and auto-filled. I guess we decided here to continue training (not fail), be fault tolerant but warn the user too.

Reply:
So one could argue, if it happens so rarely, we could as well raise. But I don't have a strong opinion on this one. Think we do this in a lot of places (just warn and continue).